### PR TITLE
Allow tooltip rotation and improve alignment of rotated elements

### PR DESCRIPTION
**Pull request by josh-ksr**
With these changes it is possible to rotate tooltips (#260). 

I also changed the alignment of rotated elements, so that all rotated elements have the same distance to the chart.

Rotated Tooltips:

Alignment including these changes:
after

For comparison: Alignment of rotated elements before, **without** these changes:
before

### CODE REVIEW

**Comment by martinkong0806** [comment]
image

The background colour when rotate still seems  off

**To reproduce**
example/lib/bar_chart/samples/bar_chart_sample3.dart line 25 - 27
image

**Comment by josh-ksr** [comment]
Thanks for testing, I'll reopen the pull request when that's fixed.

**Comment by josh-ksr** [comment]
@martinkong0806 the background problem when rotating the tooltip should be fixed with my latest changes:

File: lib/src/chart/bar_chart/bar_chart_data.dart [hunk 1]
```diff
@@ -674,6 +678,7 @@ class BarTouchTooltipData with EquatableMixin {
         fitInsideHorizontally = fitInsideHorizontally ?? false,
         fitInsideVertically = fitInsideVertically ?? false,
         direction = direction ?? TooltipDirection.auto,
+        rotateAngle = rotateAngle ?? 0.0,
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_data.dart:681]: Please add `rotateAngle` in props (see 6-7 lines below)

**Comment by imaNNeoFighT** [comment]
@josh-ksr Everything looks good and amazing. I think everyone is going to love it.

File: lib/src/chart/bar_chart/bar_chart_painter.dart [hunk 2]
```diff
@@ -358,13 +359,8 @@ class BarChartPainter extends AxisChartPainter<BarChartData> {
...
           x -= tp.width + leftTitles.margin;
           y -= tp.height / 2;
-          canvasWrapper.save();
-          canvasWrapper.translate(x + tp.width / 2, y + tp.height / 2);
-          canvasWrapper.rotate(radians(leftTitles.rotateAngle));
-          canvasWrapper.translate(-(x + tp.width / 2), -(y + tp.height / 2));
-          y -= translateRotatedPosition(tp.width, leftTitles.rotateAngle);
-          canvasWrapper.drawText(tp, Offset(x, y));
-          canvasWrapper.restore();
+          x += calculateRotationOffset(tp.size, leftTitles.rotateAngle).dx;
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_painter.dart:362]: I like these changes. makes our code simpler.
May I ask you to do this for all of our chart painters? (Wherever we duplicated this logic)

File: lib/src/chart/bar_chart/bar_chart_data.dart [hunk 3]
```diff
@@ -642,6 +642,9 @@ class BarTouchTooltipData with EquatableMixin {
   /// Controls showing tooltip on top or bottom, default is auto.
   final TooltipDirection direction;
 
+  /// Controls the rotation of the tooltip.
+  final double rotateAngle;
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_data.dart:646]: May I ask you to add rotateAngle for other tooltips? (LineTouchTooltipData and ScatterTouchTooltipData).

File: lib/src/chart/bar_chart/bar_chart_painter.dart (hunk 2 above)
Comment (josh-ksr) [reply lib/src/chart/bar_chart/bar_chart_painter.dart:362]: Unfortunately, i won't be able to do it with this pull request but possibly in the future.

**Pull request merged**